3.cellprofiling/featurization_utils/errors.py

```diff
@@ -0,0 +1,12 @@
+class ProcessorTypeError(Exception):
+    """
+    Exception raised when an unrecognized processor type is encountered.
+    Use 'CPU' or 'GPU' as valid processor types.
+    """
+
+    def __str__(self):
+        return (
+            "Processor type not recognized. Use 'CPU' or 'GPU'. "
+            "Though the dev on this project really should not allow for the code to reach this point "
+            "without the validation of a valid processor type. My apologies for the oversight."
```

Comment:
Personally I feel these comments might be too apologetic; We strove to create software which solves problems in our world - I feel it's more than fair that we raise this exception and inform the user (and also that we provided the exception to begin with). Consider removing this part.